ENGAGEHF/Managers/MessageManager/MessageManager.swift

```diff
@@ -63,17 +67,74 @@ final class MessageManager: Manager {
         if let account, account.signedIn {
             updateSnapshotListener(for: account.details)
         }
+        
+        startProcessingCleanupTimer()
+    }
+    
+    private func startProcessingCleanupTimer() {
+        activeProcessingTimer?.cancel()
+        activeProcessingTimer = Task { @MainActor in
+            while !Task.isCancelled {
+                cleanupExpiredProcessingStates()
+                try? await Task.sleep(nanoseconds: 1_000_000_000) // Check every second
+            }
+        }
     }
     
+    @MainActor
+    private func cleanupExpiredProcessingStates() {
+        processingStates = processingStates.filter { $0.value.isStillProcessing }
+        
+        // Update messages that might have expired processing states
+        for index in messages.indices {
+            if let state = messages[index].processingState, !state.isStillProcessing {
+                messages[index].processingState = nil
+            }
+        }
+    }
     
+    @MainActor
+    func markAsProcessing(type: ProcessingState.ProcessingType) {
+        let correlationId = UUID().uuidString
+        let state = ProcessingState(
+            startTime: Date(),
+            type: type,
+            correlationId: correlationId
+        )
+        
+        processingStates[correlationId] = state
+        
+        // Update all related messages' processing states
+        for index in messages.indices where messages[index].isRelatedTo(state) {
+            messages[index].processingState = state
+        }
```

Comment:
I agree with @pauljohanneskraft's assessment, I think we should not store the processing state in the message, we can keep it stored here and associate it with `MessageRow`s as they are generated in the views based in a simple function here in the message manager that will indicate if there is a processing state and that it should be observed.